app/forms/provider_interface/provider_relationship_permissions_setup_wizard.rb

```diff
@@ -73,26 +73,24 @@ def permissions_for_persistence
     # 3. check
     def next_step
       if checking_answers.present?
-        [:check]
-      elsif current_step == 'organisations'
-        [:permissions, next_provider_relationship_id]
-      elsif current_step == 'permissions' && next_provider_relationship_id.present?
-        [:permissions, next_provider_relationship_id]
-      else
-        [:check]
+        return [:check]
+      elsif current_step == 'organisations' || (current_step == 'permissions' && next_provider_relationship_id.present?)
+        return [:permissions, next_provider_relationship_id]
       end
+
+      [:check]
     end
 
     def previous_step
       if checking_answers.present?
-        [:check]
+        return [:check]
       elsif current_step == 'permissions'
-        previous_provider_relationship_id.present? ? [:permissions, previous_provider_relationship_id] : [:organisations]
+        return previous_provider_relationship_id.present? ? [:permissions, previous_provider_relationship_id] : [:organisations]
       elsif current_step == 'check'
-        [:permissions, provider_relationships.last]
-      else
-        [:check]
+        return [:permissions, provider_relationships.last]
       end
+
+      [:check]
```

Comment:
Similar to the above, is there a reason we're explicitly `return`ing in the `if`, and handling the `else` at the end of the method? I find things like this more readable:
```
if current_step == 'permissions'
  previous_provider_relationship_id.present? ? [:permissions, previous_provider_relationship_id] : [:organisations]
elsif current_step == 'check'
  [:permissions, provider_relationships.last]
else 
  [:check]
end
```

Reply:
Note my comment about `checking_answers` above.